mne/beamformer/_compute_beamformer.py

```diff
@@ -240,6 +240,10 @@ def _compute_beamformer(G, Cm, reg, n_orient, weight_norm, pick_ori,
 
     # leadfield rank and optional rank reduction
     _validate_type(reduce_rank, bool, "reduce_rank", "a boolean")
+    _check_option('inversion', inversion, ('matrix', 'single'))
+    if reduce_rank and inversion == 'single':
+        raise ValueError('reduce_rank cannot be used with inversion="single", '
+                         'use inversion="matrix" instead.')
```

Comment:
Now it sounds that the thing the user did wrong is to select the wrong inversion method, while what they actually did wrong is set `reduce_rank=True` in a case where it is probably not needed. How about:

`reduce_rank can only be used in combination with inversion="matrix"`

That leaves it open whether the user wants to change the `reduce_rank` or `inversion` parameter

Reply:
I went with a bit of a push in hopefully the right direction:
```
        raise ValueError('reduce_rank cannot be used with inversion="single"; '
                         'consider using inversion="matrix" if you have a '
                         'rank-deficient forward model (i.e., from a sphere '
                         'model with MEG channels), otherwise consider using '
                         'reduce_rank=False')
```
Sound good?

Reply:
Yes, much better!